test/topo/topotest.go

```diff
@@ -0,0 +1,134 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package topo
+
+import (
+	"context"
+	"github.com/onosproject/onos-api/go/onos/topo"
+	"github.com/onosproject/onos-config/test/utils/gnmi"
+	toposdk "github.com/onosproject/onos-ric-sdk-go/pkg/topo"
+	"github.com/stretchr/testify/assert"
+	"testing"
+	"time"
+)
+
+// NewTopoClient creates a new topo SDK client
+func NewTopoClient() (toposdk.Client, error) {
+	return toposdk.NewClient()
+}
+
+func checkTopo(t *testing.T) {
+	const (
+		onosConfigKind = topo.ONOS_CONFIG
+		devicesimKind  = "Devicesim"
+	)
+	var (
+		configEntity *topo.Entity
+		configObject topo.Object
+		targetEntity *topo.Entity
+		targetObject topo.Object
+
+		configID topo.ID
+		targetID topo.ID
+
+		relation       *topo.Relation
+		relationObject topo.Object
+	)
+
+	// Get a topology API client
+	client, err := NewTopoClient()
+	assert.NoError(t, err)
+	assert.NotNil(t, client)
+
+	// Get the list of topo objects
+	objs, err := client.List(context.Background())
+	assert.NoError(t, err)
+
+	// Find the entities for onos-config and the simulated target
+	onosConfigCount := 0
+	targetCount := 0
+
+	for _, obj := range objs {
+		if e := obj.GetEntity(); e != nil {
+			switch {
+			case e.KindID == onosConfigKind:
+				onosConfigCount++
+				configEntity = e
+				configObject = obj
+				configID = obj.ID
+			case e.KindID == devicesimKind:
+				targetCount++
+				targetEntity = e
+				targetObject = obj
+				targetID = obj.ID
+			default:
+				assert.Fail(t, "Unknown entity found in topo")
+			}
+		} else if rel := obj.GetRelation(); rel != nil {
+			relation = rel
+			relationObject = obj
+		}
+	}
+
+	// Check that there is exactly one onos-config and one simulated target
+	assert.Equal(t, 1, onosConfigCount, "wrong number of onos-config entities")
+	assert.Equal(t, 1, targetCount, "wrong number of Devicesim entities")
+
+	// Check that we found a controls relation
+	assert.NotNil(t, relation)
+	assert.Equal(t, topo.ID(topo.CONTROLS), relation.KindID)
+
+	// Check that the source and target IDs are correct
+	assert.Len(t, configEntity.SrcRelationIDs, 1)
+	assert.Len(t, targetEntity.TgtRelationIDs, 1)
+
+	configRelationID := configEntity.SrcRelationIDs[0]
+	targetRelationID := targetEntity.TgtRelationIDs[0]
+
+	assert.Equal(t, configRelationID, targetRelationID)
+	assert.Equal(t, configID, relation.SrcEntityID)
+	assert.Equal(t, targetID, relation.TgtEntityID)
+
+	// Check onos-config aspects
+	assert.Len(t, configObject.Aspects, 1)
+	aspect, ok := configObject.Aspects["onos.topo.Lease"]
+	assert.True(t, ok)
+	assert.NotNil(t, aspect)
+
+	// Check the target simulator aspects
+	assert.Greater(t, len(targetObject.Aspects), 1)
+	aspect, ok = targetObject.Aspects["onos.topo.Configurable"]
```

Comment:
Each object has a method called GetAspect you should use because it returns error and if the error is not nil then the aspect is not set.